lib/ecmascript.ts

```diff
@@ -697,34 +697,54 @@ export function ParseTemporalMonthDayString(isoString: string) {
 const TIMEZONE_IDENTIFIER = new RegExp(`^${PARSE.timeZoneID.source}$`, 'i');
 const OFFSET_IDENTIFIER = new RegExp(`^${PARSE.offsetIdentifier.source}$`);
 
+function throwBadTimeZoneStringError(timeZoneString: string): never {
+  // Offset identifiers only support minute precision, but offsets in ISO
+  // strings support nanosecond precision. If the identifier is invalid but
+  // it's a valid ISO offset, then it has sub-minute precision. Show a clearer
+  // error message in that case.
+  const msg = OFFSET.test(timeZoneString) ? 'Seconds not allowed in offset time zone' : 'Invalid time zone';
+  throw new RangeError(`${msg}: ${timeZoneString}`);
+}
+
 export function ParseTimeZoneIdentifier(identifier: string): { tzName?: string; offsetMinutes?: number } {
-  if (!TIMEZONE_IDENTIFIER.test(identifier)) throw new RangeError(`Invalid time zone identifier: ${identifier}`);
+  if (!TIMEZONE_IDENTIFIER.test(identifier)) {
+    throwBadTimeZoneStringError(identifier);
+  }
   if (OFFSET_IDENTIFIER.test(identifier)) {
-    // The regex limits the input to minutes precision
-    const { offsetNanoseconds } = ParseDateTimeUTCOffset(identifier);
+    const offsetNanoseconds = ParseDateTimeUTCOffset(identifier);
+    // The regex limits the input to minutes precision, so we know that the
+    // division below will result in an integer.
     return { offsetMinutes: offsetNanoseconds / 60e9 };
   }
   return { tzName: identifier };
 }
 
+// This operation doesn't exist in the spec, but in the polyfill it's split from
+// ParseTemporalTimeZoneString so that parsing can be tested separately from the
+// logic of converting parsed values into a named or offset identifier.
 // ts-prune-ignore-next TODO: remove if test/validStrings is converted to TS.
```

Comment:
I'm not sure I understand. AFAICT, the comment is accurate and is in the right place. In validStrings.mjs, there's this line: 

```js
const parsingMethod = ES[`ParseTemporal${mode}StringRaw`] ?? ES[`ParseTemporal${mode}String`];
```

So it's `ParseTemporalTimeZoneStringRaw` that's used in testing, and `ParseTemporalTimeZoneString` is in the spec so that comment wouldn't apply. 

@ptomato If you meant something else and I'm not understanding what you're trying to say, could you clarify? Thanks!

What this does is use the `ParseTemporalTimeZoneStringRaw` method

Reply:
Yes, you're right, I appreciate the correction.